ext/js/pages/settings/anki-deck-generator-controller.js

```diff
@@ -0,0 +1,588 @@
+                const fileName = 'anki-deck-' + today.toISOString().split('.')[0].replaceAll(/(T|:)/g, '-') + '.txt';
+                const blob = new Blob([ankiTSV], {type: 'application/octet-stream'});
+                this._saveBlob(blob, fileName);
+
+                void this._endGenerationState();
+            }, 1);
+        });
+    }
+
+    /**
+     * @param {MouseEvent} e
+     */
+    _onSendToAnki(e) {
+        e.preventDefault();
+        if (this._sendToAnkiConfirmModal !== null) {
+            this._sendToAnkiConfirmModal.setVisible(true);
+            if (this._inProgress) { return; }
+            void this._resetState();
+        }
+    }
+
+    /**
+     * @param {MouseEvent} e
+     */
+    async _onSendToAnkiConfirm(e) {
+        e.preventDefault();
+        void this._startGenerationState();
+        const terms = /** @type {HTMLTextAreaElement} */ (this._wordInputTextarea).value.split('\n');
+        const addMedia = this._addMediaCheckbox.checked;
+        const disallowDuplicates = this._disallowDuplicatesCheckbox.checked;
+        /** @type {import("anki.js").Note[]} */
+        let notes = [];
+        let index = 0;
+        requestAnimationFrame(() => {
+            this._updateProgressBar(true, 'Sending to Anki...', 0, terms.length, true);
+            setTimeout(async () => {
+                for (const value of terms) {
+                    if (!value) { continue; }
+                    if (this._cancel) {
+                        void this._endGenerationState();
+                        return;
+                    }
+                    const noteData = await this._generateNoteData(value, 'term-kanji', addMedia);
+                    if (noteData) {
+                        notes.push(noteData);
+                    }
+                    if (notes.length >= 100) {
+                        const sendNotesResult = await this._sendNotes(notes, disallowDuplicates);
+                        if (sendNotesResult === false) {
+                            void this._endGenerationStateError();
+                            return;
+                        }
+                        notes = [];
+                    }
+                    index++;
+                    this._updateProgressBar(false, '', index, terms.length, true);
+                }
+                if (notes.length > 0) {
+                    const sendNotesResult = await this._sendNotes(notes, disallowDuplicates);
+                    if (sendNotesResult === false) {
+                        void this._endGenerationStateError();
+                        return;
+                    }
+                }
+
+                void this._endGenerationState();
+            }, 1);
+        });
+    }
+
+    /**
+     * @param {import("anki.js").Note[]} notes
+     * @param {boolean} disallowDuplicates
+     * @returns {Promise<boolean>}
+     */
+    async _sendNotes(notes, disallowDuplicates) {
+        try {
+            if (disallowDuplicates) {
```

Comment:
Is this logic duplicated somewhere? Any way we can DRY (don't repeat yourself) it?

Reply:
The way duplicates are handled and notes are sent cannot match how they are handled in existing code. The logic is not the same.